libr/bin/format/pe/pe.c

```diff
@@ -2584,11 +2584,13 @@ struct PE_(r_bin_pe_obj_t)* PE_(r_bin_pe_new)(const char* file) {
 }
 
 struct PE_(r_bin_pe_obj_t)* PE_(r_bin_pe_new_buf)(struct r_buf_t *buf) {
+	RIOBind *iob = (RIOBind *)(buf->iob);
 	struct PE_(r_bin_pe_obj_t) *bin = R_NEW0 (struct PE_(r_bin_pe_obj_t));
 	if (!bin) return NULL;
 	bin->kv = sdb_new0 ();
 	bin->b = r_buf_new ();
-	bin->size = buf->length;
+	bin->size = (iob && iob->io) ? r_io_size (iob->io) : buf->length;
```

Comment:
`r_buf_size` if we update the r_buf_size, then it might break everything since actually our buffer size is buf->length but what we want to get is the file size. One workaround might be to have a function something like `r_buf_get_file_size` which returns `r_io_size(iob->io)` if `RIOBind != NULL` else returns the r_buf_size value.